DigitalLearningSolutions.Data.Tests/Services/StoreAspServiceTests.cs

```diff
@@ -0,0 +1,703 @@
+                result.validationResponse.Should().Be(TrackerEndpointResponse.StoreAspAssessException);
+                result.progress.Should().BeNull();
+                A.CallTo(() => courseDataService.GetDelegateCoursesInfo(DefaultDelegateId))
+                    .MustHaveHappenedOnceExactly();
+            }
+        }
+
+        [Test]
+        public void
+            GetProgressAndValidateCommonInputsForStoreAspProgressEndpoints_returns_progress_record_and_null_exception_when_all_is_valid()
+        {
+            // Given
+            ProgressServiceReturnsDefaultDetailedCourseProgress();
+
+            // When
+            var result = storeAspService.GetProgressAndValidateCommonInputsForStoreAspProgressEndpoints(
+                DefaultProgressId,
+                DefaultCustomisationVersion,
+                DefaultTutorialId,
+                DefaultTutorialTime,
+                DefaultTutorialStatus,
+                DefaultDelegateId,
+                DefaultCustomisationId
+            );
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.validationResponse.Should().BeNull();
+                result.progress.Should().Be(defaultCourseProgress);
+                A.CallTo(() => progressService.GetDetailedCourseProgress(DefaultProgressId))
+                    .MustHaveHappenedOnceExactly();
+            }
+        }
+
+        [Test]
+        public void
+            GetProgressAndValidateInputsForStoreAspAssess_returns_correct_progress_record_and_null_exception_when_all_is_valid()
+        {
+            // Given
+            var progressRecords = Builder<DelegateCourseInfo>.CreateListOfSize(4).All()
+                .With(p => p.RemovedDate = null)
+                .And(p => p.Completed = null)
+                .And(p => p.CustomisationId = DefaultCustomisationId)
+                .And(p => p.IsProgressLocked = false)
+                .TheFirst(1).With(p => p.RemovedDate = DateTime.UtcNow)
+                .TheNext(1).With(p => p.Completed = DateTime.UtcNow)
+                .TheNext(1).With(p => p.CustomisationId = DefaultCustomisationId + 100)
+                .Build();
+            A.CallTo(() => courseDataService.GetDelegateCoursesInfo(A<int>._)).Returns(progressRecords);
+
+            // When
+            var result = storeAspService.GetProgressAndValidateInputsForStoreAspAssess(
+                1,
+                2,
+                DefaultDelegateId,
+                DefaultCustomisationId
+            );
+
+            // Then
+            using (new AssertionScope())
+            {
+                var expectedProgressRecord = progressRecords[3];
+                result.validationResponse.Should().BeNull();
+                result.progress.Should().BeEquivalentTo(expectedProgressRecord);
+                A.CallTo(() => courseDataService.GetDelegateCoursesInfo(DefaultDelegateId))
+                    .MustHaveHappenedOnceExactly();
+            }
+        }
+

```

Comment:
It'd have been easier to review this file if the `GetProgressAndValidateInputsForStoreAspAssess` and `GetProgressAndValidateCommonInputsForStoreAspProgressEndpoints` tests were grouped separately, rather than interspersed.

Reply:
Yeah I realise that mistake now. I was trying to go with "near identical tests for different methods are together" that would make the whole file easier to understand, but from a comparison perspective it ends up worse. I've left them how they were for now so comparing between commits should be simple.